crates/bevy_ecs/src/query/world_query.rs

```diff
@@ -40,12 +40,16 @@ use variadics_please::all_tuples;
 /// [`QueryFilter`]: crate::query::QueryFilter
 pub unsafe trait WorldQuery {
     /// The item returned by this [`WorldQuery`]
-    /// For `QueryData` this will be the item returned by the query.
+    /// For `QueryData` this will be the data retrieved by the query,
+    /// and is visible to the end user when calling e.g. `Query<Self>::get`.
+    ///
     /// For `QueryFilter` this will be either `()`, or a `bool` indicating whether the entity should be included
     /// or a tuple of such things.
+    /// Archetypal query filters (like `With`) set this to `()`, as the filtering is done by selecting the archetypes to iterate over,
```

Comment:
Great! i might even add that the archetypal query filters rely on `matches_component_set` while the non-archetypal rely on `filter_fetch`

Reply:
> archetypal query filters rely on `matches_component_set` while the non-archetypal rely on `filter_fetch`

Non-archetypal filters rely on *both*, right?  `Changed<T>` filters to archetypes that contain a `T`.